reference-implementation/to-upstream-wpts/writable-streams/bad-strategies.js

```diff
@@ -4,6 +4,89 @@ if (self.importScripts) {
   self.importScripts('/resources/testharness.js');
 }
 
+const error1 = new Error('a unique string');
+error1.name = 'error1';
+
+test(() => {
+  assert_throws(error1, () => {
+    new WritableStream({}, {
+      get size() {
+        throw error1;
+      },
+      highWaterMark: 5
+    });
+  }, 'construction should re-throw the error');
+}, 'Writable stream: throwing strategy.size getter');
+
+promise_test(t => {
+  const ws = new WritableStream({}, {
+    size() {
+      throw error1;
+    },
+    highWaterMark: 5
+  });
+
+  const writer = ws.getWriter();
+
+  const p1 = promise_rejects(t, error1, writer.write('a'), 'write should reject with the thrown error');
+
+  const p2 = promise_rejects(t, error1, writer.closed, 'closed should reject with the thrown error');
+
+  return Promise.all([p1, p2]);
+}, 'Writable stream: throwing strategy.size method');
+
+promise_test(t => {
+  const sizes = [NaN, -Infinity, Infinity, -1];
+  return Promise.all(sizes.map(size => {
+    let theError;
+    const ws = new WritableStream({}, {
+      size() {
+        return size;
+      },
+      highWaterMark: 5
+    });
+
+    const writer = ws.getWriter();
+
+    return Promise.all([
+      promise_rejects(t, new RangeError(), writer.write('a').catch(r => {
```

Comment:
Use promise_rejects is just making this more complicated and harder to understand, so I suggest doing something closer to the original. For example (untested):

```javascript
return writer.write('a').then(
    () => assert_unreached('write should not fulfill');
    r => {
      assert_equals(r.name, 'RangeError', `write should reject with a RangeError for ${size}`);
      let theError = r;
      return writer.closed.then(
          () => assert_unreached('closed should not fulfill'),
          e => assert_equals(e, theError, `closed should reject with the same error for ${size}`));
      ));
```

This removes the implicit test that the write() promise fulfills before the closed promise.  If you are feeling energetic you could add an explicit test of that fact.